ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageVideoButtonsView.swift

```diff
@@ -0,0 +1,194 @@
+  var tappedCancelButton: (() -> Void)?
+
+  private let playPauseButtonImage = BluredImageView().then {
+    $0.alpha = 0
+  }
+
+  private let cancelButton = CancelButton()
+
+  init() {
+    super.init(frame: .zero)
+
+    backgroundColor = .clear
+
+    addSubview(cancelButton)
+    addSubview(playPauseButtonImage)
+
+    let tapGesture = UITapGestureRecognizer(
+      target: self,
+      action: #selector(self.videoBackgroundTapped(sender:))
+    )
+    tapGesture.numberOfTapsRequired = 1
+    addGestureRecognizer(tapGesture)
+
+    cancelButton.snp.makeConstraints {
+      $0.top.equalTo(self.safeAreaLayoutGuide.snp.top).offset(20)
+      $0.right.equalTo(self.safeAreaLayoutGuide.snp.right).offset(-20)
+    }
+    playPauseButtonImage.snp.makeConstraints {
+      $0.centerX.equalToSuperview()
+      $0.centerY.equalToSuperview().offset(20)
+    }
+  }
+
+  override func layoutSubviews() {
+    super.layoutSubviews()
+
+    cancelButton.snp.remakeConstraints {
+      $0.top.equalTo(self.safeAreaLayoutGuide.snp.top).offset(20)
+      $0.right.equalTo(self.safeAreaLayoutGuide.snp.right).offset(-20)
+    }
+  }
+
+  private func tappedVideoCancelButton() {
+    tappedCancelButton?()
+  }
+
+  @objc private func videoBackgroundTapped(sender: UITapGestureRecognizer) {
+    let location = sender.location(in: self)
+    if let view = super.hitTest(location, with: nil), view == cancelButton {
+      tappedVideoCancelButton()
+      return
+    }
+
+    guard let playStarted = tappedVideoBackground?() else {
+      return
+    }
+
+    if playStarted {
+      playPauseButtonImage.setImage(imageName: "leo.play.circle")
+      showAndFadeOutImage(imageView: playPauseButtonImage)
+    } else {
+      playPauseButtonImage.setImage(imageName: "leo.pause.circle")
+      showAndFadeOutImage(imageView: playPauseButtonImage)
+    }
+  }
+
+  private func showAndFadeOutImage(imageView: UIView) {
+    imageView.alpha = 0
+    UIView.animate(
+      withDuration: 0.1,
+      animations: {
+        imageView.alpha = 1
+      },
+      completion: { _ in
+        UIView.animate(
+          withDuration: 0.3,
+          delay: 0.5,
+          animations: {
+            imageView.alpha = 0
+          }
+        )
+      }
+    )
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+}
+
+extension NewTabPageVideoButtonsView {
+  private class CancelButton: SpringButton {
+    let imageView = UIImageView(
+      image: UIImage(braveSystemNamed: "leo.close", compatibleWith: nil)!
+    ).then {
+      $0.tintColor = .white
+      $0.contentMode = .scaleAspectFit
+    }
+
+    private let backgroundView = UIVisualEffectView(
+      effect: UIBlurEffect(style: .systemThinMaterialDark)
+    ).then {
+      $0.clipsToBounds = true
+      $0.isUserInteractionEnabled = false
+    }
+
+    override init(frame: CGRect) {
+      super.init(frame: frame)
+
+      clipsToBounds = true
+
+      addSubview(backgroundView)
```

Comment:
Please add a `accessibilityLabel = Strings.CancelString` to this init method somewhere so VoiceOver can read it